File: c-and-cpp/1907.cpp
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

void paint_blank_sequence(vector<vector<char>>& image, int r, int c, pair<int, int> start) {
    int rows = image.size();
    int cols = image[0].size();
    queue<pair<int, int>> q;
    q.push(start);

    while (!q.empty()) {
        auto current = q.front();
        q.pop();
        int row = current.first;
        int col = current.second;

        if (image[row][col] == '.') {
            image[row][col] = 'o';

            if (col + 1 < cols && image[row][col + 1] == '.') {
                q.push({row, col + 1});
            }
            if (col - 1 >= 0 && image[row][col - 1] == '.') {
                q.push({row, col - 1});
            }
            if (row + 1 < rows && image[row + 1][col] == '.') {
                q.push({row + 1, col});
            }
            if (row - 1 >= 0 && image[row - 1][col] == '.') {
                q.push({row - 1, col});
            }
        }
    }
}

int main() {
    int r, c;
    cin >> r >> c;

    vector<vector<char>> image(r, vector<char>(c));
    for (int i = 0; i < r; ++i) {
        for (int j = 0; j < c; ++j) {
            cin >> image[i][j];
        }
    }

    int paint_count = 0;

    for (int i = 0; i < r; ++i) {
        for (int j = 0; j < c; ++j) {
            if (image[i][j] == '.') {
                paint_blank_sequence(image, r, c, {i, j});
                ++paint_count;
            }
        }
    }

    cout << paint_count << endl;

    return 0;
}

File: python/1907.py
import time
from collections import deque

start = time.time()

r, c = map(int, input().split())
image = [list(input()) for _ in range(r)]


def paint_blank_sequence(image, start, r, c):
    queue = deque([start])
    while queue:
        row, col = queue.popleft()  # BFS using queue
        if image[row][col] == ".":
            image[row][col] = "o"
            if col + 1 < c and image[row][col + 1] == ".":
                queue.append((row, col + 1))
            if col - 1 >= 0 and image[row][col - 1] == ".":
                queue.append((row, col - 1))
            if row + 1 < r and image[row + 1][col] == ".":
                queue.append((row + 1, col))
            if row - 1 >= 0 and image[row - 1][col] == ".":
                queue.append((row - 1, col))


paint_count = 0

for idxr in range(r):
    for idxc in range(c):
        if image[idxr][idxc] == ".":
            paint_blank_sequence(image, (idxr, idxc), r, c)
            paint_count += 1


print(paint_count)

end = time.time()
print(end - start)
